crates/model/src/signature.rs

```diff
@@ -35,9 +36,19 @@ impl From<QuoteSigningScheme> for SigningScheme {
 #[derive(Eq, PartialEq, Clone, Deserialize, Serialize, Hash)]
 #[serde(into = "JsonSignature", try_from = "JsonSignature")]
 pub enum Signature {
+    /// Transaction data hashed and signed according to EIP-712.
+    /// https://eips.ethereum.org/EIPS/eip-712
     Eip712(EcdsaSignature),
+    /// A standard Ethereum transaction signature.
     EthSign(EcdsaSignature),
+    /// Transaction data signed according to EIP-1271, which facilitates a way for contracts to
+    /// verify signatures using an arbitrary method. This allows smart contracts to sign and
+    /// place orders.
+    /// https://eips.ethereum.org/EIPS/eip-1271
     Eip1271(Vec<u8>),
+    /// Data signed using EIP-191. Similar to the [`Self::Eip712`] variant, except the data is not
+    /// structured.
+    /// https://eips.ethereum.org/EIPS/eip-191
     PreSign,
 }
```

Comment:
Something that might make this a bit confusing is that we use each variant scheme in a way that is customized to our protocol for signing an order.
The closest to the spec is `Eip712`: we see the order as a struct and sign this struct as specified by EIP-712.
`EthSign` and `Eip1271` don't allow you to sign a struct but a bytestring and 32 bytes respectively. So what we do is looking at how EIP-712 is implemented and take the 32-byte hash that is internally used to uniquely identify the order struct and sign that with both schemes (that's what we call order hash by the way).
Finally, we also let the user tell us onchain that an order is valid: this is `PreSign` and it's done by sending a transaction onchain that contains the (EIP-712-derived) order hash as well.